src/tree.rs

```diff
@@ -444,6 +444,49 @@ impl<T> Tree<T> {
         Result::Ok(())
     }
 
+    pub fn replace(&mut self, node_id: &NodeId, mut new_node: Node<T>) -> Result<Node<T>, NodeIdError>
+    {
+        use std::mem;
+
+        let (is_valid, error) = self.is_valid_node_id(node_id);
+        if !is_valid {
+            return Result::Err(error.expect("Tree::replace: Missing an error value on finding an invalid NodeId."));
+        }
+
+        // lets cheat a bit and return new_node ;)
+        // TODO, while this is elegant, this worksaround NodeBuilder's Parameters
+        // Should we drain the old children array instead?
+        mem::swap(self.get_mut_unsafe(node_id).data_mut(), new_node.data_mut());
```

Comment:
I just realized this as well. I think the actual use case is also already possible, because you may already do just `mem::swap(node.data_mut(), &mut some_new_data)`.

I am not sure anymore, if this library should actually provide `replace`.


My original idea was to move a leaf into the tree, which would work fine for existing nodes. But I quickly realized, that there is no good way to handle interior nodes this way, because it is not clear, what happens to their children in that case. So I moved to replacing with a new node, which defeats the original point somewhat.

What I wanted to achieve can already be done much better by using `swap` and then any of the `remove` functions.

So whats left of `replace` is a shortcut for `mem::swap`, which is not that useful anymore.

If you like `replace`, I would rather implement it on `Node` directly as in `Node::replace_data(&mut self, data: T) -> T`.